lib/services/ngsi/ngsiService.js

```diff
@@ -106,6 +107,80 @@ function generateNGSIOperationHandler(operationName, entityName, typeInformation
     };
 }
 
+/**
+ * Generate an operation handler for NGSIv2-based operations (query and update). The handler takes care of identifiying
+ * the errors and calling the appropriate callback with a success or a failure depending on how the operation ended.
+ *
+ * Most of the parameters are passed for debugging purposes mainly.
+ *
+ * @param {String} operationName        Name of the NGSI operation being performed.
+ * @param {String} entityName           Name of the entity that was the target of the operation.
+ * @param {Object} typeInformation      Information about the device the entity represents.
+ * @param {String} token                Security token used to access the entity.
+ * @param {Object} options              Object holding all the information about the HTTP request.
+
+ * @return {Function}                  The generated handler.
+ */
+function generateNGSI2OperationHandler(operationName, entityName, typeInformation, token, options, callback) {
+    return function(error, response, body) {
+        if (error) {
+            logger.error(context, 'Error found executing ' + operationName + ' action in Context Broker: %s', error);
+
+            alarms.raise(constants.ORION_ALARM, error);
+            callback(error);
+        } else if (body && body.orionError) {
+            logger.debug(context, 'Orion error found executing ' + operationName + ' action in Context Broker: %j',
+                body.orionError);
+
+            callback(new errors.BadRequest(body.orionError.details));
+        } else if (response &&
+            operationName === 'update' && (response.statusCode === 204)) {
+            logger.debug(context,
+                'Received the following request from the CB:\n\n%s\n\n', JSON.stringify(body, null, 4));
```

Comment:
I think this log.debug statement is causing this log traces (seen during e2e integration tests):

```
time=2018-05-10T16:21:19.010Z | lvl=DEBUG | corr=1fe935fd-adf9-405f-9af5-e0758df643df | trans=1fe935fd-adf9-405f-9af5-e0758df643df | op=IoTAgentNGSI.NGSIService | srv=foo| subsrv=/bar| msg=Received the following request from the CB:

undefined

 | comp=IoTAgent
```

given that in 204 No Conent case, there isn't any body and the `body` variable is undefined.

If I'm rigth, this doesn't hurt too much, but it would be great to solve it, for the sake of log output cleaness.

Reply:
Fixed in 51f4e00371143d068e3f9e3549141beec73f25ac